pkg/autoscaler/autoscaler.go

```diff
@@ -106,9 +121,34 @@ func (agg *perPodAggregation) aggregate(concurrency float64) {
 	agg.probeCount += 1
 }
 
+func (agg *perPodAggregation) lameduck(t *time.Time) {
+	if agg.lameduckTime == nil {
+		agg.lameduckTime = t
+	}
+	if agg.lameduckTime.After(*t) {
+		agg.lameduckTime = t
+	}
+}
+
 // Calculates the average concurrency over all values given
-func (agg *perPodAggregation) calculateAverage() float64 {
-	return agg.accumulatedConcurrency / float64(agg.probeCount)
+func (agg *perPodAggregation) calculateAverage(now time.Time) float64 {
+	if agg.probeCount == 0 {
+		return 0.0
+	}
+	avg := agg.accumulatedConcurrency / float64(agg.probeCount)
+	if agg.lameduckTime == nil {
+		return avg
+	}
+	return agg.weightedPodCount(now) * avg
+}
+
+// Calculates the weighted pod count
+func (agg *perPodAggregation) weightedPodCount(now time.Time) float64 {
```

Comment:
Does it make sense to name this `usageRatio` or something similar? It'll always be between 0.0 and 1.0 and `weightedPodCount` keeps throwing me off when reading through.

Reply:
Yes, that's a better name.  Done.